Feat/describe instance types filtering

felipemarinho97 opened the pull request:
Added a small subset of filters as documented here: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeInstanceTypes.html

bblommers reviewed (changes requested):
A lot of the filtering logic already exists, and is part of the `moto.ec2.utils.generic_filter()` method.

A good example of this can be found in how Subnets are filtered by.
The Subnet-class itself contains the logic on what filter-key corresponds to which value:

https://github.com/spulec/moto/blob/78b89684157857d6f288de280d8f1e94e46a1bc7/moto/ec2/models/subnets.py#L125:L155

And in the backend, in `describe_instance_types`, you can just call `generic_filter`:

https://github.com/spulec/moto/blob/78b89684157857d6f288de280d8f1e94e46a1bc7/moto/ec2/models/subnets.py#L358:L359

felipemarinho97 commented:
Good suggestion! I've updated the code, the only problem is that I needed to transform the instance dictionary into an object

> A lot of the filtering logic already exists, and is part of the `moto.ec2.utils.generic_filter()` method.
> 
> A good example of this can be found in how Subnets are filtered by. The Subnet-class itself contains the `get_filter_value` method with logic on what filter-key corresponds to which value:
> 
> https://github.com/spulec/moto/blob/78b89684157857d6f288de280d8f1e94e46a1bc7/moto/ec2/models/subnets.py#L125
> 
> And in the backend, in `describe_instance_types`, you can just call `generic_filter`, which will take care of the rest (like how it's done in `SubnetBackend.get_all_subnets`).


Good suggestion! I've updated the code, the only catch is that I needed to transform the instance dictionary into an object to be able to implement `get_filter_value`

bblommers reviewed (approved):
LGTM! Thanks for adding this feature @felipemarinho97
